### PR TITLE
Fix token exchange tests

### PR DESCRIPTION
**Pull request by alexeykazakov**
- Fix token exchange tests
- Don't require redirect url param in exchange method

### CODE REVIEW

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-253-1`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-253-2`

File: controller/token.go [hunk 1]
```diff
@@ -42,10 +41,11 @@ type TokenController struct {
 	Configuration                LoginConfiguration
 	keycloakExternalTokenService keycloak.KeycloakExternalTokenService
 	providerConfigFactory        link.OauthProviderFactory
+	InitTenant                   func(ctx context.Context) error
```
Comment (sbose78) [inline controller/token.go]: We don't need this, right?

Comment (alexeykazakov) [reply controller/token.go]: No, we don't. Let me clean this up.

Comment (alexeykazakov) [reply controller/token.go]: Done https://github.com/fabric8-services/fabric8-auth/pull/253/commits/035e4c8656bdf2fefc09873c4bea422f8d26cc0d

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-253-3`

**Pull request merged**